doc/GOALS.md

```diff
@@ -72,18 +95,52 @@ them, everything else is an extension.
 
 ## 4. Goal: Performance
 
-Accessing configuration has impact on bootup and startup-time.
+Accessing configuration has impact on bootup and startup-time
+of applications.
 Elektra needs to be similar fast then current solutions.
 Ideally it should get faster because of centralized optimization
 endeavours where everyone using Elektra can benefit from.
 
-Only pay for what you need.
+> Only pay for what you need.
+
+# Domain-Specific Goals
+
+These goals are specific to the Elektra's domain, again in order of importance.
+Again, lower goals need to be ignored if goals are in conflict.
+
+## Suitability for Applications
+
+Elektra must be easy and robust for application (developers) to store any configuration they
```

Comment:
This is basically about **whom to please first**, I clarified in 285e6b65c1a5655665da4549f1c2e13124031c93

> It seems more likely that somebody will find counter examples regarding this goal.

What do you mean?

Reply:
> This is basically about whom to please first

Ok, maybe I misunderstood the intention then.

> What do you mean?

It may not be so important, but I meant that _"Elektra must be easy and robust [...] to store any configuration"_ is impossible to achieve. One direct contradiction is that we do not want to support anything that does not fit into key-value semantics.

As I said, I do agree with the idea here. It's just the wording that is used. Any configuration is a really broad spectrum. Maybe we just want to be the best at key-value?

Reply:
Ahh, now I see to which wording you refer. Yes, "any configuration" is too general here, "any key-values" would be better. Please feel free to adapt for the release.

Reply:
I'll create a PR improving the terminology.

Reply:
Like I said, maybe it is not so important if only I misunderstood it.

Reply:
No, you were right, it was easy to misunderstand and the terminology was a bit sloppy, I created the PR now: #3527

If it is still unclear, please comment there.